src/RemoteAPI/class-remote-api-base.php

```diff
@@ -80,11 +82,14 @@ public function get_api_url( array $query ): string {
 	 * Gets items from the specified endpoint.
 	 *
 	 * @since 3.2.0
+	 * @since 3.7.0 Added $associative param.
 	 *
 	 * @param array<string, mixed> $query The query arguments to send to the remote API.
+	 * @param bool                 $associative When TRUE, returned objects will be converted into associative arrays.
+	 *
 	 * @return WP_Error|array<string, mixed>
 	 */
-	public function get_items( array $query ) {
+	public function get_items( $query, $associative = false ) {
```

Comment:
I'm not seeing the value of this new arg. Is conversion to associative array truly needed? Even if it is, can this method be left simple and that conversion done in the spot where it is needed?

Reply:
Actually on PHPStan I have looked on how to give types to `stdClass` but found nothing so therefore now I tried to turn `stdClass` to `array` and then use proper typing for it.

Code wise we don't need this but for strong typing we need this so I chose the second one.